lib/json_adapter.js

```diff
@@ -1,9 +1,10 @@
 'use strict';
 
-var jsonpath = require('jsonpath-plus')
-  , minilog = require('minilog')
+var minilog = require('minilog')
   , _s = require('underscore.string');
 
+var jsonpath = false;
```

Comment:
How about:

```
var jsonpath;
try {
  jsonpath = require('jsonpath-plus');
} catch (e) {
  // could not require jsonpath-plus, this might happen when using Traverson with react-native.
  // maybe log a message like "Could not require jsonpath-plus, JSONPath support is not available."
}
```

Then add a check for `jsonpath` being truthy to `_testJSONPath`.

This might also come in handy: https://stackoverflow.com/a/43066705

But before we start shenanigans like this, I really would like to understand better what prevents jsonpath-plus from just being used in react-native.

Reply:
The stackoverflow link was a good find. It will work when adding `"react-native": { "jsonpath-plus": false },` to traversons package.json. I added it with b6e9034.

The reason jsonpath-plus is failing on react-native is that react-native does not include the module `vm` and the [is node check](https://github.com/s3u/JSONPath/blob/master/lib/jsonpath.js#L14) in jsonpath-plus will be true on react-native as well.
Another issue is that when building the bundle for react-native the bundler will not recognize the try-catches or ifs and will fail when resolving modules not available in react-native. This is fixable by adding the `react-native` property in package.json of jsonpath-plus.

I will open a PR in jsonpath-plus but considering the last commit is from beginning of January 2017 I guess it will not be addressed. This is why I started workin on traverson in the first place.